Assets/Thirdweb/Examples/Scripts/Prefab_ConnectWallet.cs

```diff
@@ -0,0 +1,256 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Thirdweb;
+using System;
+using TMPro;
+
+// Can be added to SDK
+public enum Chain
+{
+    Ethereum,
+    Goerli,
+    Polygon,
+    Mumbai,
+    Arbitrum,
+    ArbitrumGoerli,
+    Avalanche,
+    AvalancheFujiTestnet,
+    BinanceSmartChainMainnet,
+    BinanceSmartChainTestnet,
+    Fantom,
+    FantomTestnet,
+    Optimism,
+    OptimismGoerli,
+}
+
+// Can be added to SDK
+[Serializable]
+public enum Wallet
+{
+    MetaMask,
+    CoinbaseWallet,
+    WalletConnect,
+    Injected,
+    MagicAuth,
+}
+
+// Can be replaced with Serializeable Dictionary (asset)
+[Serializable]
+public struct WalletButton
+{
+    public Wallet wallet;
+    public GameObject walletButton;
+}
+
+public class Prefab_ConnectWallet : MonoBehaviour
+{
+    [Header("SETTINGS")]
+    public Chain chain;
+    public List<Wallet> supportedWallets;
+
+    [Header("UI - CONNECTING")]
+    public GameObject connectButton;
+    public Transform connectDropdown;
+    public List<WalletButton> walletButtons;
+
+    [Header("UI - CONNECTED")]
+    public GameObject connectedButton;
+    public Transform connectedDropdown;
+    public TMP_Text connectInfoText;
+    public TMP_Text walletAddressText;
+
+    bool dropdownExpand;
+    Coroutine dropdownRoutine;
+    string address;
+
+    ThirdwebSDK SDK;
+
+    // SDK Initialization
+
+    private void Awake()
+    {
+#if !UNITY_EDITOR
+        SDK = new ThirdwebSDK(chain.ToString().ToLower());
```

Comment:
this wont work with chains names that have a `-` like `optimism-goerli`. One thing that might help for this is that we're about to merge a PR in the TS SDK that now accepts ChainID as the network input. So that will work nicely with an enum. For now I think passing string is safer

Reply:
Can you point me to where I can find the expected strings? But yes chain ID as the network input would solve it.